File: typescript/prime-factors/prime-factors.ts
export default function calculatePrimeFactors(n: number) {
    const factors = []
    let factor = 2

    while (n > 1) {
        while (n % factor !== 0) {
            factor += factor === 2 ? 1 : 2
        }

        n /= factor
        factors.push(factor)
    }

    return factors
}

File: ruby/prime-factors/prime_factors.rb
module PrimeFactors
  def self.of(number)
    factor = 2
    factors = []

    while number > 1
      factor += factor == 2 ? 1 : 2 while number.remainder(factor).positive?
      number /= factor
      factors << factor
    end

    factors
  end
end
